messaging/usecase/room_usecase.go

```diff
@@ -0,0 +1,111 @@
+package usecase
+
+import (
+	"chat/domain"
+	"chat/utils/errors"
+	"fmt"
+	"log"
+	_ "reflect"
+	"time"
+)
+/*
+
+	SaveRoom( room *ChatRoom) error
+	GetChatRoomsFor( studentID string) (*StudentChatRooms, error)
+	EditChatRoomParticipants( room *ChatRoom) error
+	DeleteRoom( userID string, roomID string) error
+*/
+type roomUseCase struct {
+	rr      domain.RoomRepository
+	st 		domain.StudentRepository
+	timeout time.Duration
+}
+
+func (u *roomUseCase) SaveRoom(room *domain.ChatRoom) error {
+
+	_,err := u.rr.GetRoom(room.RoomID)
+	if err==nil {
+		return errors.NewConflictError(fmt.Sprintf("Room with ID %s already exists", room.RoomID))
+	}
+	_,err=u.st.GetStudent(room.Admin.ID)
+	if err==nil {
+		return errors.NewConflictError(fmt.Sprintf("The admin with ID  %s does not exist", room.Admin.ID))
+	}
+	err = u.rr.SaveRoom( room )
```

Comment:
after SaveRoom, we need to CreateStudentRooms bc we have 2 tables; chat.room and chat.student_rooms, that we have to update whenever a room is created or deleted